tests/functional/src/main/java/org/mule/functional/classloading/isolation/maven/MavenArtifact.java

```diff
@@ -32,14 +32,13 @@
     private final String scope;
 
     /**
-     * Creates a maven representation of an artifact. The identity of a {@link MavenArtifact} is based on
-     * its not null attributes, version and scope don't define the identity.
+     * Creates a maven representation of an artifact
      *
      * @param groupId not null
      * @param artifactId not null
      * @param type not null
      * @param version can be null
-     * @param scope can be null
+     * @param scope not null
```

Comment:
In this case, these params should be non empty (ie: not enough to be not null). Add the corresponding validation in code